packages/table/src/locator.ts

```diff
@@ -122,8 +156,22 @@ export class Locator implements ILocator {
         return bodyRect.left + this.grid.getCumulativeWidthAt(index);
     }
 
+    private convertCellMidpointToClientX = (index: number) => {
+        const bodyRect = this.getBodyRect();
+        const cumWidth = this.grid.getCumulativeWidthAt(index);
+        const prevCumWidth = (index > 0) ? this.grid.getCumulativeWidthAt(index - 1) : 0;
```

Comment:
getCumulative width is fairly expensive because it does a linear scan. Instead of doing it twice, you should just get the width of the specific cell from the grid.